src/audio-worklet/design-pattern/lib/wasm-audio-helper.js

```diff
@@ -215,20 +217,23 @@ class RingBuffer {
 
     // Transfer data from the |arraySequence| storage to the internal buffer.
     const sourceLength = arraySequence[0].length;
+    const dataArray = this._data;
+    const channelCount = this._channelCount;
+    const length = this._length;
+
     for (let i = 0; i < sourceLength; ++i) {
-      for (let channel = 0; channel < this._channelCount; ++channel) {
-        this._channelData[channel][this._writeIndex] = arraySequence[channel][i];
+      for (let channel = 0; channel < channelCount; ++channel) {
+        const index = (this._writeIndex + i) % length;
```

Comment:
Did you measure that this is more efficient than storing back to the object member?  There was a recent change that did the opposite of what is being proposed here so it would be good to quantify which version is faster, and maybe add a comment with the results as well.